fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RemoteFetchTest.java

```diff
@@ -292,15 +307,97 @@ void testReadYourWriteOutOfRangeSucceeds(RecordQueryPlannerConfiguration.IndexFe
+    void failAfterRecordsReturnedTest(RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod) throws Exception {
+        assumeTrue(recordStore.getContext().isAPIVersionAtLeast(APIVersion.API_VERSION_7_1));
+
+        List<TestRecords1Proto.MySimpleRecord> created = saveManyRecords();
+
+        // Modify record and then scan unmodified index
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, splitRecordsHook);
+
+            // Update a record. This record will eventually be returned in a query, but we do *not* modify
+            // a field in the index being scanned, so that FDB does not detect the range conflict until later
+            TestRecords1Proto.MySimpleRecord lastRecord = created.get(created.size() - 1);
+            recordStore.saveRecord(lastRecord.toBuilder()
+                    .setStrValueIndexed("foo")
+                    .build());
+
+            // Use remote fetch to scan the num_value_unique index. The first few results should return
+            // data (essentially the first few pages of data from scanning the index), but once it
+            // gets to the final page, it should fail because it sees a modified range when trying to look
+            // up the record
+            RecordQueryPlan plan = plan(NUM_VALUES_LARGER_EQUAL_0, fetchMethod);
+            if (fetchMethod == RecordQueryPlannerConfiguration.IndexFetchMethod.USE_REMOTE_FETCH) {
+                assertThrows(ExecutionException.class, () -> executeToList(context, plan, null, ExecuteProperties.SERIAL_EXECUTE));
+            } else {
+                executeAndVerifyData(context, plan, null, ExecuteProperties.SERIAL_EXECUTE, 500, (rec, i) -> {
+                    int primaryKey = i;
+                    int numValue = i;
+                    String strValue = (i == created.size() - 1) ? "foo" : "";
+                    assertRecord(rec, primaryKey, strValue, numValue, "MySimpleRecord$num_value_unique", (long)numValue);
+                });
+            }
+        }
+    }
+
+    @ParameterizedTest(name = "failAfterRecordsReturnedReverseTest(" + ARGUMENTS_WITH_NAMES_PLACEHOLDER + ")")
+    @EnumSource()
+    void failAfterRecordsReturnedReverseTest(RecordQueryPlannerConfiguration.IndexFetchMethod fetchMethod) throws Exception {
+        assumeTrue(recordStore.getContext().isAPIVersionAtLeast(APIVersion.API_VERSION_7_1));
+
+        List<TestRecords1Proto.MySimpleRecord> created = saveManyRecords();
+
+        // Modify record and then scan unmodified index
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, splitRecordsHook);
+
+            // Update a record. This record will eventually be returned in a query, but we do *not* modify
+            // a field in the index being scanned, so that FDB does not detect the range conflict until later
+            TestRecords1Proto.MySimpleRecord lastRecord = created.get(created.size() - 1);
```

Comment:
```suggestion
            TestRecords1Proto.MySimpleRecord lastRecord = created.get(0);
```
I think for the reverse test, it should edit the first result, as that lets it scan a bunch of results first before hitting the one in the RYW cache